test/test_crud_v2.py

```diff
@@ -0,0 +1,331 @@
+# Copyright 2015-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test the collection module."""
+
+import json
+import os
+import re
+import sys
+
+sys.path[0:0] = [""]
+
+from bson.py3compat import iteritems
+from pymongo import operations, WriteConcern
+from pymongo.command_cursor import CommandCursor
+from pymongo.cursor import Cursor
+from pymongo.errors import PyMongoError
+from pymongo.read_concern import ReadConcern
+from pymongo.results import _WriteResult, BulkWriteResult
+
+from test import unittest, client_context, IntegrationTest
+from test.utils import OvertCommandListener, drop_collections, rs_client
+from test.utils_selection_tests import parse_read_preference
+
+# Location of JSON test specifications.
+_TEST_PATH = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), 'crud', 'v2')
+
+# Default test database and collection names.
+TEST_DB = 'testdb'
+TEST_COLLECTION = 'testcollection'
+
+
+def camel_to_snake(camel):
+    # Regex to convert CamelCase to snake_case.
+    snake = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', camel)
+    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', snake).lower()
+
+
+def camel_to_upper_camel(camel):
+    return camel[0].upper() + camel[1:]
+
+
+def camel_to_snake_args(arguments):
+    for arg_name in list(arguments):
+        c2s = camel_to_snake(arg_name)
+        arguments[c2s] = arguments.pop(arg_name)
+    return arguments
+
+
+def parse_collection_options(opts):
+    if 'readPreference' in opts:
+        opts['read_preference'] = parse_read_preference(
+            opts.pop('readPreference'))
+
+    if 'writeConcern' in opts:
+        opts['write_concern'] = WriteConcern(
+            **dict(opts.pop('writeConcern')))
+
+    if 'readConcern' in opts:
+        opts['read_concern'] = ReadConcern(
+            **dict(opts.pop('readConcern')))
+    return opts
```

Comment:
It would be great if you could move `camel_to_snake`, `camel_to_upper_camel`, `camel_to_snake_args`, and `parse_collection_options` into `test/utils.py` so the various spec runners could reuse them.